### ISSUE TITLE
Add Custom Pickup Date Options to Diaper Request Page

### ISSUE THREAD

**Issue by pdxdiaperbank** [rubyforgood/partner#104]
Diaper Banks preschedule specific dates each month for their partner agencies to pickup diapers. Some diaper banks even create an advance calendar for the _entire year._  See our 2019 agency pickup schedule here: [2019 Diaper Distribution Schedule.pdf](https://github.com/rubyforgood/partner/files/3025660/2019.Diaper.Distribution.Schedule.pdf)

For that reason, it's important for diaper banks to be able to input those pickup date options into the system, so when partner agencies place their diaper requests each month via ParterBase, they can select their choice on what date to pick up their diapers. 

As a result, everyone is on the same page in regards to scheduling-- Partner agencies know when to mark their calenders for pickup, and diaper banks now have an accurate deadline for fulfilling/preparing diapers orders for pickup. 

**The dream:** For the system to add partner agencies into the DiaperBase calendar system based upon their selected pickup date. 

I hope that all makes sense! Please let me know if you have any questions. Thanks, everyone!

**Comment by Nirvikalpa108** [comment]
I've spoken to the wonderful Megan Fischer 😃  about this and understand the following: 

- Diaperbanks can input their pickup schedule so that partners know when they can pick up diapers.
- The schedule consists of dates (Monday 29 July 2019) and a time slot (eg 0930-1130).
-  Partners can select their chosen pickup slot so that the Diaperbank will know when they are coming to pick up diapers.
- Partners will receive email notifications reminding them to visit the Diaperbank at their chosen date and time.

This is important, because:
- Diaperbanks may have limited inventory space to store diapers, so if a Partner does not show up at their allotted date/time, they may have to unpack their order and re-pack them for another day.
- Some partners miss their scheduled pickup slot and the families that would have received those diapers may miss out on having diapers for their children.

Actions for our engineers ❤️:
- Create this issue in Diaperbase
- Create a way for Diaperbanks to input their pickup schedule in Diaperbase
- Partnerbase GETs this schedule from Diaperbase
- Partners can select their pickup time as part of the Diaper request user journey 
- Partners receive an email notification before their chosen pickup date/time (the app could check for these every Friday for the following week for example).

**Comment by benreyn** [comment]
Thanks so much for the discovery work on this @Nirvikalpa108 and for writing all this up! cc: @pdxdiaperbank, @scdb. If either of you have anything to add, please let us know!

**Comment by SCDB** [comment]
Perfection!! You captured my brain perfectly. :)

**Comment by pdxdiaperbank** [comment]
Yes!! Perfectly captured. Thanks for getting the ball rolling on this.
